src/output.c

```diff
@@ -917,7 +972,7 @@ static int JsonGenericLogger(ThreadVars *tv, void *thread_data, const Packet *p,
     }
 
     const char *name;
-    switch (al->proto) {
+    switch (f->proto) {
```

Comment:
instead of doing this at runtime, can we do it at init time and store the result in `EveJsonSimpleAppLayerLogger`?

Reply:
nice idea :-)